### ISSUE TITLE
[BUG] Engine fails to load if an 'ace' object already exists on the page

### ISSUE THREAD

**Issue by tombrunet** [IBMa/equal-access#331]
Affects automated tools. For example:
```
<html>
<head>
<script type="text/javascript">
    var ace = {}
</script>
</head>
<body>
...
</body>
```

On these pages, the engine will think it's already loaded and will fail to load into the page. We need to think through a more robust means of engine loading in these scenarios.

**Comment by rstedman** [comment]
This is a problem for any page making use of the fairly popular [ace](https://ace.c9.io/) embeddable code editor. The ace library also adds a global ace object, preventing any pages that use this component from having tests written that make use of accessibility-checker.

This has caused issues for some projects at my company. We've been using accessibility-checker for over a year with protractor and jasmine, and have been found it very valuable. Recently we've pulled in a component which makes use of ace, which caused all of our accessibility tests to start failing.

For now I've been able to work around this by running `await browser.executeScript('delete window.ace;');` before calling `aChecker.getCompliance`, but it isn't really ideal.

**Comment by tombrunet** [comment]
Seems to also happen when any page has an element with `id="ace"`. https://codesandbox.io/s/issue-331-id-gdb5ud

**Comment by tombrunet** [comment]
Upping the priority since this blocks checking in random situations without obvious workarounds.

**Comment by shunguoy** [comment]
This issue can happen in different cases:
  1.  when an element with an id = 'ace' in the page, the extension crashed with:   
<img width="951" alt="Screen Shot 2022-09-15 at 10 24 35 AM" src="https://user-images.githubusercontent.com/18539498/190447463-e799db6a-0a2c-46ac-ae6d-eb16a9b0c458.png">
 this issue doesn't occur in the command line or the automated test.  it seems extension-specific.
 2.  when the accessibility namespace "ace" colids with any other user library
 3.  when the "ace' object is already loaded

**Comment by shunguoy** [comment]
extension version and test case:
<img width="405" alt="Screen Shot 2022-09-15 at 10 53 16 AM" src="https://user-images.githubusercontent.com/18539498/190450433-d1a589d5-5cbd-4700-b928-6f12c631b43e.png">

```
<html lang="en">
<body>
<div aria-grabbed="true" style="border: 2px solid" id="ace">
       element with aria-grabbed=true
</div>
<div aria-dropeffect="copy" style="border: 2px solid" >
        element with aria-dropeffect=copy
</div>
</body>
</html>
```

**Comment by shunguoy** [comment]
explore different options to fix issues.

**Comment by shunguoy** [comment]
when an object is defined as global, the object will always exist, and can not be 'undefined'.  The script engine (like puppeteer) will use the one as long as it's defined even though it is nulled.

**Comment by shunguoy** [comment]
fixed w/ the pull request #1086

**Comment by eirckr** [comment]
Testing with https://codesandbox.io/s/issue-331-id-gdb5ud and is not scanning this page @shunguoy @drjoho please review: 
<img width="857" alt="Screen Shot 2022-09-27 at 1 46 36 PM" src="https://user-images.githubusercontent.com/62436670/192610982-b43b892d-2d00-4c0e-93a0-ef318186d92e.png">
<img width="790" alt="Screen Shot 2022-09-27 at 1 46 44 PM" src="https://user-images.githubusercontent.com/62436670/192610985-6c00fba4-a512-4d35-85bd-47b3f7808c82.png">

**Comment by shunguoy** [comment]
@eirckr  what commands you used to run the page?  steps to duplicate?

**Comment by eirckr** [comment]
1. Navigate to test url: https://codesandbox.io/s/issue-331-id-gdb5ud 
2. Run the checker from [action ](https://github.com/IBMa/equal-access/actions/runs/3137984865)

Url does not scan

**Comment by eirckr** [comment]
Validated
